pkg/ebpf/ingress_node_firewall_events.go

```diff
@@ -28,20 +31,20 @@ func (infc *IngNodeFwController) ingressNodeFwEvents() {
 	// described in the eBPF C program.
 	rd, err := perf.NewReader(objs.IngressNodeFirewallEventsMap, os.Getpagesize())
 	if err != nil {
-		log.Printf("Failed creating perf event reader: %q", err)
-		return
+		return fmt.Errorf("Failed creating perf event reader: %q", err)
 	}
-	logFile, ok := os.LookupEnv("EVENT_LOGGING_FILE")
-	if !ok {
-		log.Printf("EVENT_LOGGING_FILE env variable must be set")
-		return
-	}
-	file, err := os.OpenFile(logFile, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
-	if err != nil {
-		log.Printf("Cannot open events logging file at path %q: %v", logFile, err)
-		return
+
+	var eventsLogger *syslog.Writer
+
+	if err := wait.PollImmediate(time.Second, 30*time.Second, func() (done bool, err error) {
+		if eventsLogger, err = syslog.New(syslog.LOG_INFO|syslog.LOG_DAEMON, "daemon"); err != nil {
+			log.Printf("failed to connect to syslog: %v; Retrying...", err)
```

Comment:
and remove `daemon` from here too

Reply:
If you make this `""` then the syslog package will default to `os.Args[0]`. 

```go
// New establishes a new connection to the system log daemon. Each
// write to the returned writer sends a log message with the given
// priority (a combination of the syslog facility and severity) and
// prefix tag. If tag is empty, the os.Args[0] is used.
func New(priority Priority, tag string) (*Writer, error) {
	return Dial("", "", priority, tag)
}

// Dial establishes a connection to a log daemon by connecting to
// address raddr on the specified network. Each write to the returned
// writer sends a log message with the facility and severity
// (from priority) and tag. If tag is empty, the os.Args[0] is used.
// If network is empty, Dial will connect to the local syslog server.
// Otherwise, see the documentation for net.Dial for valid values
// of network and raddr.
func Dial(network, raddr string, priority Priority, tag string) (*Writer, error) {
	if priority < 0 || priority > LOG_LOCAL7|LOG_DEBUG {
		return nil, errors.New("log/syslog: invalid priority")
	}

	if tag == "" {
		tag = os.Args[0]
	}
	hostname, _ := os.Hostname()

	w := &Writer{
		priority: priority,
		tag:      tag,
		hostname: hostname,
		network:  network,
		raddr:    raddr,
	}

	w.mu.Lock()
	defer w.mu.Unlock()

	err := w.connect()
	if err != nil {
		return nil, err
	}
	return w, err
}
```

As we have our own syslog, dropping it from `logParts` should be sufficient.

Leaving this as "daemon" and dropping `logParts["tag"]` gives us:

```console
% oc logs -c events -f -n ingress-node-firewall-system ingress-node-firewall-daemon-vkxff
2022-08-19 16:55:08 +0000 UTC kind-worker2 ruleId 10 action Drop len 98 if eth0
2022-08-19 16:55:08 +0000 UTC kind-worker2 	ipv4 src addr 172.18.0.1 dst addr 172.18.0.4
2022-08-19 16:55:08 +0000 UTC kind-worker2 	icmpv4 type 8 code 0
2022-08-19 16:55:09 +0000 UTC kind-worker2 ruleId 10 action Drop len 98 if eth0
2022-08-19 16:55:09 +0000 UTC kind-worker2 	ipv4 src addr 172.18.0.1 dst addr 172.18.0.4
2022-08-19 16:55:09 +0000 UTC kind-worker2 	icmpv4 type 8 code 0
2022-08-19 16:55:10 +0000 UTC kind-worker2 ruleId 10 action Drop len 98 if eth0
2022-08-19 16:55:10 +0000 UTC kind-worker2 	ipv4 src addr 172.18.0.1 dst addr 172.18.0.4
2022-08-19 16:55:10 +0000 UTC kind-worker2 	icmpv4 type 8 code 0
2022-08-19 16:55:11 +0000 UTC kind-worker2 ruleId 10 action Drop len 98 if eth0
2022-08-19 16:55:11 +0000 UTC kind-worker2 	ipv4 src addr 172.18.0.1 dst addr 172.18.0.4
2022-08-19 16:55:11 +0000 UTC kind-worker2 	icmpv4 type 8 code 0
2022-08-19 16:55:12 +0000 UTC kind-worker2 ruleId 10 action Drop len 98 if eth0
2022-08-19 16:55:12 +0000 UTC kind-worker2 	ipv4 src addr 172.18.0.1 dst addr 172.18.0.4
2022-08-19 16:55:12 +0000 UTC kind-worker2 	icmpv4 type 8 code 0
```